File: codeeval/lettercase_percentage_ratio.py
import sys

with open(sys.argv[1], "r") as f:
    for line in f:
        upper, line = 0.0, line.rstrip()
        for c in line:
            if c.isupper():
                upper += 1
        print "lowercase: %2.2f uppercase: %2.2f" % \
              (((len(line) - upper) / len(line))*100, (upper / len(line)) * 100)

File: codeeval/lettercase_percentage_ratio.rb
File.open(ARGV[0]).each_line do |line|
    up = line.chomp.each_char.count {|c| c == c.upcase}.to_f
    printf("lowercase: %.2f uppercase %.2f \n", 
            ((line.size - up) / line.size) * 100, 
            (up / line.size) * 100)
end
